File: app/src/main/java/app/islammedia/halaqatalquran/Database/Student.java
package app.islammedia.halaqatalquran.Database;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity
public class Student {
    @NonNull
    public String getIdName() {
        return idName;
    }

    public void setIdName(@NonNull String idName) {
        this.idName = idName;
    }

    @ColumnInfo(name = "idName") @PrimaryKey @NonNull
    private String idName;

    @ColumnInfo(name = "name") @NonNull
    private String name;

    @ColumnInfo(name = "moreInfo")
    private String info;





    @NonNull
    public String getName() {
        return name;
    }

    public void setName(@NonNull String name) {
        this.name = name;
    }

    public String getInfo() {
        return info;
    }

    public void setInfo(String info) {
        this.info = info;
    }

    public int getHalaqaID() {
        return halaqaID;
    }

    public void setHalaqaID(int halaqaID) {
        this.halaqaID = halaqaID;
    }

    @ColumnInfo(name = "hisHalaqa")
    private int halaqaID;



    public Student(@NonNull String name, String info, @NonNull String idName) {
        this.name = name;
        this.info = info;

        this.idName = idName;
    /*;*/
    }
}

File: app/src/main/java/app/islammedia/halaqatalquran/database/entities/Student.kt
package app.islammedia.halaqatalquran.database.entities

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity
class Student(@field:ColumnInfo(name = "name") private var studentName: String,
              @field:ColumnInfo(name = "moreInfo") private val info: String?,
              @field:PrimaryKey @field:ColumnInfo(name = "idName") private var studentId: String)
    : Comparable<Student> {
    fun getStudentId(): String {
        return studentId
    }

    fun setStudentId(studentId: String) {
        this.studentId = studentId
    }

    fun getName(): String {
        return studentName
    }

    fun setName(name: String) {
        this.studentName = name
    }

    fun getInfo(): String? {
        return info
    }

    fun getHalaqaID(): Int {
        return halaqaID
    }

    fun setHalaqaID(halaqaID: Int) {
        this.halaqaID = halaqaID
    }

    @ColumnInfo(name = "hisHalaqa")
    private var halaqaID = 0

    override fun compareTo(other: Student): Int {
        return when {
            this.getName() < other.studentName -> -1
            this.getName() > other.studentName -> 1
            else -> 0
        }
    }
}